src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionHelper.cs

```diff
@@ -100,12 +113,18 @@ public static string GetExpressionText(LambdaExpression expression)
                 nameParts.Pop();
             }
 
+            expressionText = string.Empty;
             if (nameParts.Count > 0)
             {
-                return nameParts.Aggregate((left, right) => left + right).TrimStart('.');
+                expressionText = nameParts.Aggregate((left, right) => left + right).TrimStart('.');
```

Comment:
Earlier comment on this got buried. Making this better is part of the next wave of changes